Description: Use default handler for SIGPIPE to avoid broken pipe errors
 because Rust ignores SIGPIPE by default [1], and println! panics on error [2],
 like when stdout is closed. [1] shows a `println_loop | head -n1` example,
 which is exactly what tests in tests/choose.rs use as the "chooser" command.
 This should fix #1078430, but has to be observed for a while to be sure.
 .
 [1]: https://github.com/rust-lang/rust/issues/46016
 [2]: https://github.com/BurntSushi/advent-of-code/issues/17
Last-Update: 2024-08-10
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/tests/test.rs
+++ b/tests/test.rs
@@ -62,2 +62,4 @@
   pub(crate) fn new() -> Self {
+    /* SAFETY: this is the default in C. */
+    unsafe { libc::signal(libc::SIGPIPE, libc::SIG_DFL); }
     Self::with_tempdir(tempdir())
